### ISSUE TITLE
iOS Storage Unknown Error on download when file exists

### ISSUE THREAD

**Issue by traxes9k** [aws-amplify/amplify-swift#2607]
### Describe the bug

There is an unhandled error when trying to download a file that already exists with the iOS SWIFT SDK. Attaching the logs .

### Steps To Reproduce

```swift
1. Download a file with Storage SDK
2. Download a file with Storage SDK to the same path
```

### Expected behavior

Either overwrite the file or return a specific error that can be caught. Best case scenario would be to implement a feature that would allow to overwrite when S3 Last modified > local item Last Modified.

### Amplify Framework Version

2.2.0

### Amplify Categories

Storage

### Dependency manager

Swift PM

### Swift version

5.7

### CLI version

-

### Xcode version

14.1

### Relevant log output

```shell
music.png
Progress: <NSProgress: 0x6000033ad480> : Parent: 0x0 (portion: 0) / Fraction completed: 1.0000 / Completed: 5465 of 5465  
StorageError: Unexpected error occurred with message: An unknown error occurred
Recovery suggestion: This should not happen. There is a possibility that there is a bug if this error persists. Please take a look at https://github.com/aws-amplify/amplify-ios/issues to see if there are any existing issues that match your scenario, and file an issue with the details of the bug if there isn't. Issue encountered at:
file: /Users/{USER_PLACEHOLDER_FOR_PRIVACY}/Library/Developer/Xcode/DerivedData/Askyno-avosbeohlxrhvdfpkfpgrvdfulwz/SourcePackages/checkouts/amplify-swift/Amplify/Categories/Storage/Error/StorageError.swift
function: recoverySuggestion
line: 63
Caused by:
fatalError(errorDescription: "File already exists at destination: /Users/{USER_PLACEHOLDER_FOR_PRIVACY}/Library/Developer/CoreSimulator/Devices/BC9E3CD6-E56A-41E5-8BD1-2F0FED372BAE/data/Containers/Data/Application/C2E23916-3C63-4E18-AE63-58D118C92073/Documents/music.png")
```

### Is this a regression?

No

### Device

iPhone 14 Pro-Simulator

### iOS Version

16.1

### Specific to simulators

NO

**Comment by phantumcode** [comment]
@traxes9k Thank you for submitting the issue.  I was able to reproduce the error.

The Storage category currently does not support downloading and overwriting to an existing file.  We will on improving this behavior.

In order to help us understand the usages better, can you provide the use case that is causing you to download to the same file/file location? What is the time interval in between the first download and subsequent download?

**Comment by phantumcode** [comment]
Issue is fixed and released in version [2.6.1](https://github.com/aws-amplify/amplify-swift/releases/tag/2.6.1).
